utils/awsutils/session.go

```diff
@@ -9,70 +9,98 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
 	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/mitchellh/mapstructure"
 )
 
 type SessionConfig struct {
-	Region      string `mapstructure:"region"`
-	AccessKeyID string `mapstructure:"accessKeyID"`
-	AccessKey   string `mapstructure:"accessKey"`
-	IAMRoleARN  string `mapstructure:"iamRoleARN"`
-	ExternalID  string `mapstructure:"externalID"`
-	Service     string
-	Timeout     time.Duration
+	Region           string        `mapstructure:"region"`
+	AccessKeyID      string        `mapstructure:"accessKeyID"`
+	AccessKey        string        `mapstructure:"accessKey"`
+	IAMRoleARN       string        `mapstructure:"iamRoleARN"`
+	ExternalID       string        `mapstructure:"externalID"`
+	Endpoint         *string       `mapstructure:"endpoint"`
+	S3ForcePathStyle *bool         `mapstructure:"s3ForcePathStyle"`
+	DisableSSL       *bool         `mapstructure:"disableSSL"`
+	Service          string        `mapstructure:"service"`
+	Timeout          time.Duration `mapstructure:"timeout"`
 }
 
 func createRoleSessionName(serviceName string) string {
 	return fmt.Sprintf("rudderstack-aws-%s-access", strings.ToLower(serviceName))
 }
 
-func createDefaultSession(config *SessionConfig) *session.Session {
-	return session.Must(session.NewSession(&aws.Config{
+func createDefaultSession(config *SessionConfig) (*session.Session, error) {
```

Comment:
No value for `error` is returned from the function.

Reply:
session.NewSession returns (response, error) so I am just passing that.